### PR TITLE
Signup Improvements/Bugs

### PR DESCRIPTION
**Pull request by kmichael500**
- after signup, it takes you to an email confirmation screen to make it clear that you need to confirm your email
- after confirming your email, on the same browser, it will log you in automatically if it is within 30 min of signup
- fixed weird signup bugs
- Mailgun email web hooks
- Client routes are defined in one place and get generated
- backend routes genearted
- sensitive data redacted from db logs

### CODE REVIEW

File: Planarian/Planarian/Modules/Account/Services/AccountUserManagerService.cs [hunk 1]
```diff
@@ -67,48 +77,54 @@ public async Task<string> InviteUser(InviteUserRequest request, CancellationToke
...
-
-            await _emailService.SendAccountInvitationEmail(user, accountUser,
-                accountName);
-            await dbTransaction.CommitAsync(cancellationToken);
-            
-            return user.Id;
         }
-        catch (Exception)
+
+        var sendResult = await _emailService.SendAccountInvitationEmail(
```
Comment (kmichael500) [inline Planarian/Planarian/Modules/Account/Services/AccountUserManagerService.cs:119]: [P1] Preserve successful invitation creation across post-commit email failures

At this point the temporary `User`/`AccountUser` (and any default permissions) have already been committed at line 107. `SendAccountInvitationEmail` can still throw outside its provider-send catch—for example while creating the `MessageLog`/template/message or while persisting submission state—and line 120 can also throw after Mailgun accepted the message. The controller then returns an error, the UI leaves the invite flow in its failure path, and retrying can create another temporary `User`/`AccountUser` because the email uniqueness constraint excludes temporary users. That can leave duplicate active invitations/permissions, potentially after the first email was actually sent.

Once the creation transaction commits, this endpoint needs to treat creation as authoritative: isolate post-commit email/tracking failures and return the created invitation with a failed/unknown delivery state, or otherwise make the operation idempotent/durably queued so a retry cannot create a second invitation.

File: Planarian/Planarian/Modules/Users/Services/UserService.cs [hunk 2]
```diff
@@ -118,34 +118,44 @@ public async Task RegisterUser(RegisterUserVm user, CancellationToken cancellati
...
             {
-                await ClaimInvitation(entity, user.InvitationCode, cancellationToken);
+                await dbTransaction.RollbackAsync(cancellationToken);
+                throw;
             }
+        }
 
-            await _emailService.SendEmailConfirmationEmail(entity.EmailAddress, entity.FullName,
-                entity.EmailConfirmationCode);
+        var sendResult = await _emailService.SendEmailConfirmationEmail(
```
Comment (kmichael500) [inline Planarian/Planarian/Modules/Users/Services/UserService.cs:148]: [P1] Don't turn a committed signup into a failed registration

The user (and invitation claim, when present) is committed at line 139 before this call. `SendEmailConfirmationEmail` converts provider `Send()` failures into `EmailSendResult.SendFailed`, but setup/logging/status-persistence failures can still escape. If one does, `RegisterController` never gets a successful result, so it never issues the registration-continuation cookie; the frontend reports registration failure even though the account exists, and a retry hits `EmailAlreadyExists`.

The new pending-screen `SendFailed` path therefore does not cover every post-commit failure. Once registration commits, return account creation successfully with failed/unknown confirmation-delivery state (and allow the continuation cookie to be issued), while logging/reconciling email-tracking failures separately.

File: Planarian.Web/src/Shared/Components/EmailHistoryModal/EmailHistoryModal.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              {attempt.events.length === 0 ? (
+                <Text type="secondary">No provider events recorded yet.</Text>
+              ) : (
+                <div className="email-history-modal__events">
+                  {attempt.events.map((event, eventIndex) => (
+                    <Text
+                      type="secondary"
+                      key={`${attempt.messageLogId}-${event.occurredOn}-${eventIndex}`}
+                    >
+                      {event.eventType} · {formatDateTime(event.occurredOn)}
```
Comment (kmichael500) [inline Planarian.Web/src/Shared/Components/EmailHistoryModal/EmailHistoryModal.tsx]: [P3] Render event enums through a display map

This generic user-facing modal already maps `MessageDeliveryStatus` through `MessageDeliveryStatusDisplay`, but event types are rendered as the raw wire enum here, so values such as `PermanentFailed` / `TemporaryFailed` leak directly into the UI. Please give `MessageDeliveryEventType` the same dedicated display-dictionary treatment and use it here. That keeps serialized enum values decoupled from user-facing copy and keeps the generic modal consistent when reused.

**Review by kmichael500 — COMMENTED** [review]
I reviewed the full PR against `main`, including the registration/confirmation continuation flow, invitation lifecycle, Mailgun tracking/webhooks, cookie migration, generated route contracts, migration/redaction logic, UI behavior, and tests. The overall direction is strong, but I found two post-commit consistency bugs that can turn a successfully committed operation into an apparent request failure and then produce bad retry behavior. I consider those blocking before merge. I also left one smaller user-facing enum-display consistency comment.

File: Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+                                END;
+
+                                route_path := regexp_replace(url_without_query, '^https?://[^/]+', '', 'i');
+
+                                IF lower(route_path) IN ('/confirm-email', '/reset-password') AND query_string IS NOT NULL THEN
+                                    query_string := regexp_replace(
+                                        query_string,
+                                        '(^|&)(code)=([^&]*)',
+                                        E'\1\2=[redacted]',
+                                        'gi');
```
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs:55]: I checked this against PostgreSQL's string-literal and `regexp_replace` rules, and this is a false positive. The migration is a C# raw string, so PostgreSQL receives `E'\\1\\2=[redacted]'`. Inside an `E'...'` literal, `\\` produces one literal backslash; `regexp_replace` therefore receives `\1\2=[redacted]`, which is exactly the capture-backreference form it expects. PostgreSQL's own docs use `E'X\\1Y'` as the canonical example for a replacement backreference. The invitation replacement below uses the same correct escaping. Adding another level of backslashes here would change the replacement semantics rather than fix them.

Refs: https://www.postgresql.org/docs/current/sql-syntax-lexical.html and https://www.postgresql.org/docs/15/functions-matching.html

File: Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+using Planarian.Model.Shared;
+
+namespace Planarian.Shared.Routing;
+
+public sealed class InvitationCodeRouteConstraint : LengthRouteConstraint
+{
+    public InvitationCodeRouteConstraint() : base(PropertyLength.InvitationCode)
+    {
+    }
+}
```
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs:11]: I don't think this matches the actual contract in this PR. `InvitationCodeRouteConstraint` is deliberately a named/source-of-truth replacement for the previous `{code:length(10)}` constraint; the tests assert `PropertyLength.InvitationCode` and fixed-length routing, not rejection of punctuation. I also ran the full `Planarian.Tests.EmailDelivery` project and all 136 tests pass, so the claimed contract-test failure does not occur.

Generated invitation codes are alphanumeric, but the API route did not previously enforce that alphabet. An arbitrary 10-character value simply reaches the lookup and fails as an unknown invitation. If we want route-level alphabet validation as extra hardening, that should be an explicit contract change with a corresponding test rather than being treated as a regression in this PR.

File: Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx [hunk 6]
```diff
@@ -34,6 +39,7 @@ const LoginPage: React.FC = () => {
 
   let redirectUrl = "/";
   if (!isNullOrWhiteSpace(encodedRedirectUrl)) {
+    // TODO: Validate redirectUrl as a local Planarian path before navigating to prevent open redirects.
     redirectUrl = decodeURIComponent(encodedRedirectUrl as string);
   }
```
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx:44]: This is worth hardening, but it is not introduced by #313. `main` already reads `redirectUrl`, decodes it, and passes it to `navigate(redirectUrl)` after login; this PR only adds the TODO beside that existing behavior (the invitation navigation change is separate). I would track/fix the redirect validation as an existing security-hardening issue rather than treat it as a regression or merge blocker for this signup/email PR.

File: Planarian/Planarian/Shared/Email/Services/MailgunWebhookService.cs [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    public MailgunWebhookService(EmailOptions emailOptions, MessageLogRepository messageLogRepository)
+    {
+        _emailOptions = emailOptions;
+        _messageLogRepository = messageLogRepository;
+    }
+
+    public async Task<MailgunWebhookProcessingResult> Process(MailgunWebhookVm? payload,
+        CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(_emailOptions.WebhookSigningKey))
```
Comment (kmichael500) [inline Planarian/Planarian/Shared/Email/Services/MailgunWebhookService.cs:35]: [P2] Fail fast on missing webhook configuration instead of relying on `Retry`

`WebhookSigningKey` is required to authenticate this endpoint, but the app can start with it unset and then every webhook returns the `Retry`/503 path here. Mailgun explicitly documents that non-200/406 webhook responses are retried **except for the delivery notification**, so a misconfigured deployment can permanently lose delivery events while the application otherwise boots healthy. The same concern applies to a missing configured domain below.

Please validate the required Mailgun webhook configuration at hosted startup (or introduce an explicit disabled mode where the Mailgun webhook is not configured) rather than treating missing configuration as a transient request failure.

Provider behavior: https://documentation.mailgun.com/docs/mailgun/user-manual/webhooks/webhook-retries

**Review by kmichael500 — COMMENTED** [review]
One additional Mailgun deployment/reliability finding from the provider-doc validation pass: missing required webhook configuration should fail fast rather than relying on a 503 retry response.

File: Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs:55]: Fixed in `e7c13c4c`.

I wrote the regression tests first and confirmed the original behavior was real: confirmation/invitation email setup failures escaped after the primary database transaction had already committed. I also added coverage for the outer URL-construction/setup boundary, which was another way the same post-commit failure could occur.

The fix now treats confirmation/invitation email setup/tracking failures as delivery failures instead of failing the already-committed primary operation. If the provider has already accepted the message, later local bookkeeping failure preserves `Submitted` rather than misreporting the send as failed. Invitation `InvitationSentOn` persistence is also best-effort after provider submission so it cannot turn a successful invitation create into a retryable API failure.

Validation after the fix: backend email-delivery suite 144/144, frontend 77/77, release API build clean, production web build successful.

File: Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs:11]: Fixed in `e7c13c4c`.

I added the regression coverage first and reproduced the bug before touching production code: a post-commit confirmation-email setup failure could make signup throw even though the user already existed. I also found and test-covered the URL-construction boundary outside the original send helper, which could fail in the same way.

Confirmation email setup/tracking failures now return a `SendFailed` delivery result instead of escaping after account creation has committed, so the registration endpoint can still complete its successful account-creation response/continuation flow. If Mailgun already accepted the message, a later local bookkeeping failure reports `Submitted` rather than downgrading the send.

Final backend email-delivery suite: 144/144 passing.

File: Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx:44]: Fixed in `e7c13c4c`.

`MessageDeliveryEventType` now has a dedicated `MessageDeliveryEventTypeDisplay` map, and the generic email-history modal renders through it instead of showing raw wire enum values such as `PermanentFailed`. The modal regression test was changed first and failed against the original implementation (`Permanent failure` appeared only for delivery status while the event still rendered `PermanentFailed`); it now passes and explicitly verifies the raw enum text is absent.

Full frontend suite: 77/77 passing; production build successful.

File: Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs:55]: Resolved separately: this automated migration concern was a false positive, not part of `e7c13c4c`. PostgreSQL escape-string parsing leaves the intended backreference for `regexp_replace` here, so no migration change was made.

File: Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs:11]: No code change was made for this automated comment. The named constraint intentionally preserves the previous `{code:length(10)}` behavior; the existing tests do not require punctuation rejection, and the full backend suite passes. The review finding was therefore not treated as a bug.

File: Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx:44]: No fix was included for this automated comment because the redirect behavior already exists on `main`; this PR only added the TODO beside it. It is worth addressing as a separate security cleanup, but it is not a regression introduced by PR #313.

File: Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian.Migrations/Migrations/20260814140909_v30.cs:55]: Correction: this reply was intended for the original post-commit invitation finding below, not this nested review note. The fix itself is still `e7c13c4c`; I’m posting the same implementation/test details on the correct original thread as well.

File: Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian/Planarian/Shared/Routing/InvitationCodeRouteConstraint.cs:11]: Correction: this reply was intended for the original committed-signup/email-confirmation finding, not this nested review note. I’m posting it on the correct original thread as well.

File: Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (kmichael500) [reply Planarian.Web/src/Modules/Authentication/Pages/LoginPage.tsx:44]: Correction: this reply was intended for the original email-history enum-display finding, not this nested review note. I’m posting it on the correct original thread as well.

File: Planarian/Planarian/Modules/Account/Services/AccountUserManagerService.cs (hunk 1 above)
Comment (kmichael500) [reply Planarian/Planarian/Modules/Account/Services/AccountUserManagerService.cs:119]: Fixed in `e7c13c4c`.

I wrote the regression coverage first and reproduced the failure before changing production code. The tests proved that invitation email setup failures could escape after the `User`/`AccountUser` transaction had committed. During the fix pass I also verified the outer invitation-URL construction boundary and added a failing test for that before fixing it.

The email boundary now converts pre-submission setup/tracking failures into a failed delivery result instead of failing the already-committed invitation. If Mailgun has already accepted the message, later local submission-status bookkeeping failure preserves `Submitted`. The post-send `InvitationSentOn` save is also best-effort and logged so it cannot turn a successful invitation create into a retryable API failure.

Final validation: backend email-delivery suite 144/144; release API build clean.

File: Planarian/Planarian/Modules/Users/Services/UserService.cs (hunk 2 above)
Comment (kmichael500) [reply Planarian/Planarian/Modules/Users/Services/UserService.cs:148]: Fixed in `e7c13c4c`.

I added regression tests first and reproduced the committed-signup failure before touching production code. The tests also exposed a second outer boundary: confirmation URL construction could throw before reaching the protected send helper, so I added a red test for that path as well before fixing it.

Confirmation email setup/tracking failures now return `SendFailed` instead of escaping after registration has committed, allowing the registration endpoint to complete the successful account-creation/continuation flow. If Mailgun already accepted the message, later local bookkeeping failure remains `Submitted` rather than being downgraded.

Final backend email-delivery suite: 144/144 passing.

File: Planarian.Web/src/Shared/Components/EmailHistoryModal/EmailHistoryModal.tsx (hunk 3 above)
Comment (kmichael500) [reply Planarian.Web/src/Shared/Components/EmailHistoryModal/EmailHistoryModal.tsx]: Fixed in `e7c13c4c`.

The test was changed first to require the display value and to assert that raw `PermanentFailed` text is absent; it failed against the original implementation as expected. `MessageDeliveryEventType` now has a dedicated `MessageDeliveryEventTypeDisplay` dictionary, and the generic email-history modal renders event types through it.

Final frontend suite: 77/77 passing; production web build successful.

File: Planarian/Planarian/Shared/Email/Services/MailgunWebhookService.cs (hunk 7 above)
Comment (kmichael500) [reply Planarian/Planarian/Shared/Email/Services/MailgunWebhookService.cs:35]: Fixed in `e7c13c4c`.

I changed the existing webhook tests first so missing signing-key/domain configuration expected rejection instead of `Retry`; they failed against the original implementation (`Actual: Retry`). I also added a hosted-configuration validator test first, which failed to compile until the startup validation existed.

Hosted deployments now fail startup when `Email:Domain` or `Email:WebhookSigningKey` is missing. In local/development misconfiguration, the webhook service rejects the request instead of returning the misleading 503 retry path. This avoids relying on provider retries for delivery notifications, which Mailgun does not guarantee.

Final backend email-delivery suite: 144/144 passing.
